controllers/storagecluster/storagecluster_controller.go

```diff
@@ -166,13 +166,17 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	onboardingSecretPredicates := builder.WithPredicates(
-		predicate.NewPredicateFuncs(
-			func(client client.Object) bool {
-				return client.GetName() == onboardingTicketPublicKeySecretName
-			},
-		),
-	)
+	secretPredicates := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.GetName() == onboardingTicketPublicKeySecretName
```

Comment:
This is wrong as it will prevent delete events on all watched secrets not just the onboarding public key secret
You just need to return `true` here